federatedscope/core/optimizers/adadp.py

```diff
@@ -0,0 +1,76 @@
+import torch
+from torch.optim.optimizer import Optimizer
+import numpy as np
+
+class ADADP(Optimizer):
```

Comment:
I didn't see any usage of this optimizer. Which module has instantiated ADADP?

Reply:
This is an implementation of the following paper:

A. Koskela and A. Honkela. "Learning Rate Adaptation for Differentially Private Learning." In: International Conference on Artificial Intelligence and Statistics. PMLR, 2020. p. 2465-2475.